superset/utils/pandas_postprocessing/flat.py

```diff
@@ -0,0 +1,81 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pandas as pd
+
+from superset.utils.pandas_postprocessing.utils import (
+    _is_multi_index_on_columns,
+    FLAT_COLUMN_SEPARATOR,
+)
+
+
+def flat(df: pd.DataFrame, reset_index: bool = True,) -> pd.DataFrame:
```

Comment:
Just a thought (doesn't need to be done here): Should we introduce a separate operation `index` that sets the index without needing to do a full `pivot` along with aggregations on the metrics? Then we could perhaps call them `set_index` and `flatten_index` or something that clearly communicates that we're specifically changing the index.

Reply:
It seems like `stack` and `unstack` function in Pandas. I will add those in future.